app/javascript/packages/document-capture/components/submission-complete.jsx

```diff
@@ -13,14 +15,38 @@ import SubmissionInterstitial from './submission-interstitial';
 /**
  * @typedef SubmissionCompleteProps
  *
- * @prop {Resource<any>} resource Resource object.
+ * @prop {Resource<UploadSuccessResponse>} resource Resource object.
  */
 
+export class RetrySubmissionError extends Error {}
+
+/**
+ * Interval after which to retry submission, in milliseconds.
+ *
+ * @type {number}
+ */
+const RETRY_INTERVAL = 2500;
+
+/**
+ * Returns a promise resolving after given timeout.
+ *
+ * @param {number} ms Timeout in millseconds.
+ *
+ * @return {Promise<void>}
+ */
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
 /**
  * @param {SubmissionCompleteProps} props Props object.
  */
 function SubmissionComplete({ resource }) {
-  resource.read();
+  const response = resource.read();
+
+  if (response.status === 'in_progress') {
+    throw sleep(RETRY_INTERVAL).then(() => {
+      throw new RetrySubmissionError();
+    });
```

Comment:
Would we ever need to worry about this being re-entrant? Like should we have it check if there's already a sleep and then cancel that?

Reply:
I don't think so, but it'd be worth looking into. I'd expect that if React has special treatment for these thrown promises in its Suspense feature, that it would handle it (and use as a re-render trigger) via attaching its own `then` to the promise value, and hopefully not rerender any sooner. But that's just my assumption at this point.

Reply:
So it does need some additional consideration:

1. If the component is unmounted for any reason, we should want to make sure to clear the timer ([documentation](https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect))
2. Throwing an error in the `then` like this will not be caught by the error boundary as I hoped it would. This is discussed at https://github.com/facebook/react/issues/14981, with a recommended workaround ("trick") using `setState` 🤷  https://github.com/facebook/react/issues/14981#issuecomment-468460187

Updates in e1b3b5d. This also happens to be a bit cleaner in consolidating "on mount" behaviors.

I'll still want to do some additional testing on this next week.